### PR TITLE
Revamp actionpack-xml_parser gem

### PR DESCRIPTION
**Pull request by rafaelfranca**
Since Rails 5, parameter parsing is not done in a middleware. This gem was using the middleware to do the parameter parsing so it was failing.

This gem now only works with Rails 5 so we can use the new API to register a new parameter parser.

It was removed the support to legacy `HTTP_X_POST_DATA_FORMAT` to set the content type.

Review @jeremy @matthewd @eileencodes since basecamp may be using this.

### CODE REVIEW

File: lib/action_pack/xml_parser.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+module ActionPack
+  class XmlParser
+    def self.register
+      original_parsers = ActionDispatch::Request.parameter_parsers
+      ActionDispatch::Request.parameter_parsers = original_parsers.merge(Mime[:xml].symbol => self)
+    end
+
+    def self.call(raw_post)
+      Hash.from_xml(raw_post) || {}
+    end
```
Comment (kaspth) [inline lib/action_pack/xml_parser.rb]: Holy hell, this implementation got a lot simpler :D

**Comment by kaspth** [comment]
Looks 🚢'able to me, caveat is that we're not using this in our app and I'm tired, so might have missed something 😄

**Comment by eileencodes** [comment]
Tested this in our app and all the tests pass. I tested some of the endpoints as well and saw no issues. 👍

**Comment by axlekb** [comment]
Any update on the merge-ability of this PR?

**Pull request merged**